packages/expo-modules-core/ios/DynamicTypes/DynamicDictionaryType.swift

```diff
@@ -0,0 +1,65 @@
+// Copyright 2015-present 650 Industries. All rights reserved.
+
+/**
+ A dynamic type representing dictionary types. Requires the dictionary's value type
+ for the initialization as it delegates casting to that type for each element in the dictionary.
+ */
+internal struct DynamicDictionaryType: AnyDynamicType {
+  let valueType: AnyDynamicType
+
+  func wraps<InnerType>(_ type: InnerType.Type) -> Bool {
+    if let dictionaryType = InnerType.self as? AnyDictionary.Type {
+      return valueType.equals(dictionaryType.getValueDynamicType())
+    }
+    return false
+  }
+
+  func equals(_ type: AnyDynamicType) -> Bool {
+    if let dictionaryType = type as? Self {
+      return dictionaryType.valueType.equals(valueType)
+    }
+    return false
+  }
+
+  func cast(jsValue: JavaScriptValue, appContext: AppContext) throws -> Any {
+    if let jsObject = try? jsValue.asObject() {
+      var result: [AnyHashable: Any] = [:]
+      for key in jsObject.getPropertyNames() {
+        result[key] = try valueType.cast(jsValue: jsObject.getProperty(key), appContext: appContext)
+      }
+      return result
+    }
+    throw Conversions.CastingException<JavaScriptObject>(jsValue)
+  }
+
+  func cast<ValueType>(_ value: ValueType, appContext: AppContext) throws -> Any {
+    if let value = value as? [AnyHashable: Any] {
+      return try value.mapValues { try valueType.cast($0, appContext: appContext) }
+    }
+    throw Conversions.CastingException<[AnyHashable: Any]>(value)
+  }
+
+  var description: String {
+    "[AnyHashable: \(valueType.description)]"
```

Comment:
Probably we should also pass a key type to this dynamic type so the description could be more accurate. If not yet, at least let's use Hashable instead (`AnyHashable` is a struct, `Hashable` is a protocol which seems better in my opinion)
```suggestion
    "[Hashable: \(valueType.description)]"
```

Reply:
thanks! good to learn more about swift hashable.
i would still not include key type if the key type is only for description.